etl/steps/data/garden/climate/2024-01-31/ghg_concentration.py

```diff
@@ -0,0 +1,140 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import List
+
+import pandas as pd
+from owid.catalog import Table
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Columns to select from the data, and how to rename them.
+COLUMNS = {
+    "year": "year",
+    "month": "month",
+    "average": "concentration",
+    # The following column is loaded only to perform a sanity check.
+    "decimal": "decimal",
+}
+
+
+def add_rolling_average(tb: Table, original_column_names: List[str]) -> Table:
+    tb_with_average = tb.copy()
+
+    # Create a date range.
+    date_range = pd.date_range(start=tb_with_average["date"].min(), end=tb_with_average["date"].max(), freq="1D")
+
+    # Get unique locations.
+    unique_locations = tb_with_average["location"].unique()
+
+    # Set date as index and sort.
+    tb_with_average = tb_with_average.set_index(["location", "date"]).sort_index()
+
+    # Create a MultiIndex with all possible combinations of date and location.
+    multi_index = pd.MultiIndex.from_product([unique_locations, date_range], names=["location", "date"])
+
+    # Reindex using the MultiIndex.
+    tb_with_average = tb_with_average.reindex(multi_index)
+
+    for original_column_name in original_column_names:
```

Comment:
Sorry, I don't understand your question. What do you mean by "this"?

Reply:
Ah sorry I just meant the process of interpolating daily data from monthly and then calculating annual averages.

Reply:
Thanks for pointing that out. I have just committed a change to do interpolations monthly. It's deinitely not necessary to do daily interpolations. I think I just did that because I copied the function from another place, where the measurements were done on different days of the month. In fact, for this data, nothing changes, and it's not necessary to do any interpolation, since all months are informed. But in case there is a single month missing, then we need interpolation, otherwise all points after the missing one will be nan (or probably we'd need to do the calculation in a slightly different way).